src/finn/transformation/fpgadataflow/make_zynq_proj.py

```diff
@@ -499,7 +602,7 @@ def apply(self, model):
             kernel_model = ModelWrapper(dataflow_model_filename)
             # InsertFIFO at this stage interferes with tLastMarker
             # TODO: is this really needed here at all?
-            if not self.enable_instrumentation:
+            if not self.enable_instrumentation:  # or self.instrumentation_no_dma == False
```

Comment:
Is InsertFIFO() needed here for the DMA case to work correctly? Then we would have a conflict in the DMA+Instrumentation case because InsertFIFO apparently interferes with the tLastMarker (as per the code comment above).